virtcontainers/sandbox.go

```diff
@@ -2150,32 +2150,43 @@ func (s *Sandbox) cpuResources() *specs.LinuxCPU {
 
 // setupSandboxCgroup creates and joins sandbox cgroups for the sandbox config
 func (s *Sandbox) setupSandboxCgroup() error {
+	var err error
 	spec := s.GetPatchedOCISpec()
-
 	if spec == nil {
 		return errorMissingOCISpec
 	}
 
-	if spec.Linux == nil {
-		// Cgroup path is optional, though expected. If not defined, skip the setup
-		s.Logger().WithField("sandboxid", s.id).Warning("no cgroup path provided for pod sandbox, not creating sandbox cgroup")
-		return nil
+	s.state.CgroupPath, err = validCgroupPath(spec.Linux.CgroupsPath, s.config.SystemdCgroup)
+	if err != nil {
+		return fmt.Errorf("Invalid cgroup path: %v", err)
 	}
-	validContainerCgroup := utils.ValidCgroupPath(spec.Linux.CgroupsPath)
 
-	// Create a Kata sandbox cgroup with the cgroup of the sandbox container as the parent
-	s.state.CgroupPath = filepath.Join(filepath.Dir(validContainerCgroup), cgroupKataPrefix+"_"+s.id)
-	cgroup, err := cgroupsNewFunc(cgroups.V1, cgroups.StaticPath(s.state.CgroupPath), &specs.LinuxResources{})
-	if err != nil {
-		return fmt.Errorf("Could not create sandbox cgroup in %v: %v", s.state.CgroupPath, err)
+	// Do not change current cgroup configuration.
+	// Create a spec without constraints
+	unconstraintSpec := specs.Spec{
+		Linux: &specs.Linux{
+			Resources:   &specs.LinuxResources{},
+			CgroupsPath: s.state.CgroupPath,
+		},
+	}
 
+	cmgr, err := newCgroupManager(s.config.Cgroups, s.state.CgroupPaths, &unconstraintSpec)
```

Comment:
nit: in the commit message for cb60505de20553214366cb97e5cbf495a9507f25 can you include a blurb on the use of a cgroup manager (ie, that we are using a new library now?)